projects/baby_driver/src/main.c

```diff
@@ -0,0 +1,56 @@
+#include "can.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+
+#define CAN_DEVICE_ID 0x1
+
+typedef enum {
+  CAN_EVENT_RX = 0,
+  CAN_EVENT_TX,
+  CAN_EVENT_FAULT,
+  NUM_CAN_EVENTS,
+} CanEvent;
+
+static CanStorage s_can_storage;
+static CanSettings s_can_settings = {
+  .device_id = CAN_DEVICE_ID,
+  .bitrate = CAN_HW_BITRATE_500KBPS,
+  .rx_event = CAN_EVENT_RX,
+  .tx_event = CAN_EVENT_TX,
+  .fault_event = CAN_EVENT_FAULT,
+  .tx = { GPIO_PORT_A, 12 },
+  .rx = { GPIO_PORT_A, 11 },
+  .loopback = false,
+};
+
+static GpioSettings s_led_settings = { .direction = GPIO_DIR_OUT,
+                                       .state = GPIO_STATE_HIGH,
+                                       .alt_function = GPIO_ALTFN_NONE,
+                                       .resistor = GPIO_RES_NONE };
+
+static StatusCode prv_default_rx_callback(const CanMessage *msg, void *context,
+                                          CanAckStatus *ack_reply) {
+  LOG_DEBUG("Received CAN Message\n");
+  return STATUS_CODE_OK;
+}
+
+int main() {
+  LOG_DEBUG("Welcome to BabyDriver!\n");
+  gpio_init();
+  event_queue_init();
+  interrupt_init();
+
+  can_init(&s_can_storage, &s_can_settings);
+  can_register_rx_default_handler(prv_default_rx_callback, NULL);
```

Comment:
I'm pretty sure the thought was that individual babydriver modules (like gpio_set etc) would have their own RX handlers that would just get registered here, I don't think we need a default callback

Reply:
Ah, that was just for the new members so that they can have an example if they were working on it. I can just remove it though.

Reply:
Yeah, there should be examples in the ticket or the firmware 10X notes. Don't think it's necessary to clutter the code.